src/buildercore/context_handler.py

```diff
@@ -53,5 +53,18 @@ def download_from_s3(stackname, refresh=False):
     s3.download(key, expected_path)
     return True
 
+def only_if(servicename):
+    """Decorator that only executes an update function if the context contains a particular servicename that would need it"""
+    def decorate_with_only_if(fn):
+        def decorated_with_only_if(stackname, context, **kwargs):
+            # only update service if stack is using given service
+            LOG.info("Try to update '%s'", servicename)
+            if context.get(servicename):
+                # TODO: context is not always necessary in fn implementations. Can we avoid passing it when not needed?
```

Comment:
it doesn't hurt either. avoiding passing the context would make the code unnecessarily more complicated.
https://docs.python.org/3/library/inspect.html#inspect.getargspec